File: backend/web api/Models/CourseCategory.cs
public class CourseCategory
{
    public int CourseID { get; set; }
    public int MainCategoryID { get; set; }
    public int SubCategoryID { get; set; }


    public Course Course { get; set; }
    public MainCategory MainCategory { get; set; }
    public SubCategory SubCategory { get; set; }
}

File: learning-platform/src/app/models/courseCategory.ts
export interface CourseCategory {
    courseID: number;        // معرف الكورس
    mainCategoryID?: number; // التصنيف الرئيسي (اختياري)
    subCategoryID?: number;  // التصنيف الفرعي (اختياري)
  }
  